### DISCUSSION TITLE
New client for GNU/Linux: opensnitch

### DISCUSSION THREAD

**Discussion by gustavo-iniguez-goya** [badmojr/1Hosts#732] [General]
Hi!

Thank you for maintaining these lists.

I'm using these lists with OpenSnitch, so maybe it could be added to the list of clients? It supports hosts format, list of IPs, network ranges and domains with regular expressions:

https://github.com/evilsocket/opensnitch/wiki/block-lists#notes

**Comment by badmojr** [comment]
Hello.
Is hosts file the only format supported? How about formats such as this, https://o0.pages.dev/Pro/domains.txt ?

**Comment by gustavo-iniguez-goya** [comment]
Yes, it's also supported. Supported formats are: hosts, list of IPs, list of network ranges and list of domains (regular expressions allowed)

**Comment by badmojr** [comment]
Ok! Which of the formats do you personally use?

**Reply by gustavo-iniguez-goya** [reply]
hosts format mainly. I also use domains list with regular expressions, but sometimes it causes false positives.

**Comment by badmojr** [comment]
I'm not not on MacOS, so I am unable to take the app for a test drive myself.


Would u please test these formats below?
Please disable any other lists first, and send some ss or videos if possible.
https://o0.pages.dev/Pro/domains.txt
https://o0.pages.dev/Pro/domains.wildcards

**Comment by gustavo-iniguez-goya** [comment]
opensnitch is for Linux only, not mac :)

testing the list with 2 domains:

[opensnitch-1hosts.webm](https://user-images.githubusercontent.com/2742953/192171195-ba14e4cc-420a-4b85-a6c7-7f023a6a63e3.webm)

testing the list with a web browser:

[opensnitch-1hosts-1.webm](https://user-images.githubusercontent.com/2742953/192171230-330adbd0-4ef8-48f8-a304-96812fd31c41.webm)

**Comment by badmojr** [comment]
That's amazing!
Please now test with only `https://o0.pages.dev/Pro/domains.wildcards ` as the blocklist.
Do a browser test too.

**Comment by gustavo-iniguez-goya** [comment]
Done!

[opensnitch-1hosts-wildcards.webm](https://user-images.githubusercontent.com/2742953/192255813-2cb6077c-288b-4f06-b766-8030fff904ff.webm)

**Comment by badmojr** [comment]
Great! I got one last question though.
What's the difference between the `deny` and `reject` rule on the first and the last videos?

**Reply by gustavo-iniguez-goya** [reply]
Deny discards the requests, until the system timeouts the connection. On the other hand Reject actively closes the connection. If you compare both videos with Deny and Reject, notice how fast loads ebay when rejecting ads domains.

**Comment by badmojr** [comment]
Cool! It's now included.

**Reply by gustavo-iniguez-goya** [reply]
Thank you ! I hope it helps GNU/Linux users to discover this app to block ads/malware, etc system wide :)
